jax/experimental/compilation_cache/compilation_cache.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hashlib
+import jax
+
+
+def get_cache_key(xla_computation, compile_options) -> str:
+    hash_obj = hashlib.sha256()
+    hash_obj.update(xla_computation.as_serialized_hlo_module_proto())
+    _hash_compile_options(hash_obj, compile_options)
+    hash_obj.update(bytes(jax.lib.version))
+    _hash_libtpu(hash_obj, jax.lib.xla_bridge.get_backend())
+    return hash_obj.digest().hex()
+
+def _hash_compile_options(hash_obj, compile_options_obj):
+    assert (len(dir(compile_options_obj)) == 31), "The compile options are not consistent with the version you ran with for previously compiled programs."
```

Comment:
I took out the outer parenthesis too because mypy wasn't happy with them...I was getting the error "Assert is always True"...likewise I added parenthesis around the string part in order for python to be happy with the line breaks.